docs/previews.md

```diff
@@ -176,33 +177,144 @@ const AuthorTemplate = ({ location, data: staticData }) => {
+undefined or erroneous `path`, resulting unexpected behavior.
+
+To remedy this, we're going to take advantage of the fact that Gatsby pages and
+templates are just React components.
+
+#### Unpublished document route handler
+
+To start, let's create a dedicated `unpublishedPreview` page in your `pages/`
+directory. (e.g. `src/pages/unpublishedPreview.js`).
+
+> ⚠️ While not required, we recommend client rendering the `unpublishedPreview`
+> route with
+> (`gatsby-plugin-create-client-paths`)[https://www.npmjs.com/package/gatsby-plugin-create-client-paths]
+> as well. This will avoid issues with `window` during SSR.
+
+In here, we can have the following component:
+
+```js
+// => in src/pages/unpublishedPreview.js
+import { PageTemplate } from 'src/templates/page'
+
+export const UnpublishedPage = props => {
+  const previewData = window.__PRISMIC_PREVIEW_DATA__
+
+  // => Perform any logic from previewData to determine the correct page or template component to use.
+
+  return <PageTemplate {...props} />
+}
+
+export default UnpublishedPage
+```
+
+As usual, let's break this down:
+
+1. We import our `<PageTemplate>` component. Because templates and pages in
+   Gatsby are just React components, we can reuse them anywhere!
+2. Read our `previewData` from `window`. This example reads `previewData`
+   straight from `window`, but feel free to conditionally assign this in any way
+   you prefer to prevent SSR issues.
+3. Run any logic using `previewData` to correctly determine what template
+   component to pass your data to. This example only has one template, but your
+   site may utilize several.
+4. Spread your props over the template componenet you render. This is needed to
+   pass props like `location` and any other props your templates may need.
+
+#### Resolving unpublished paths
+
+Next, we need to update the `linkResolver` or `pathResolver` function that we
+use to resolve our `path` from `usePrismicPreview` to handle unpublished
+documents. Because we may need some custom logic that may not be suitable for
+your link resolving, `pathResolver` is typically more appropriate for this.
+
+Depending on the structure of your project, the implementation of `pathResolver`
+may vary, but here is a full example to get started with:
+
+```js
+// in src/pages/preview.js
+export const PreviewPage = ({ location }) => {
+  const { allPrismicPage } = useStaticQuery(graphql`
+    {
+      allPrismicPage {
+        edges {
+          node {
+            uid
+          }
+        }
+      }
+    }
+  `)
+  const pageUIDs = allPrismicPage.edges.map(node => node.uid)
+
+  const pathResolver = () => doc => {
+    const previewedUID = doc.prismicPage.uid
+
+    if (pageUIDs.includes(previewedUID)) {
+      return previewedUID
+    } else {
+      return '/unpublishedPreview'
+    }
+  }
+
+  const { previewData, path } = usePrismicPreview(location, {
+    repositoryName: 'myRepoName',
+    linkResolver: () => doc => doc.uid,
+    htmlSerializer: () => {},
```

Comment:
Possible to leave this out and use a default `() => {}` instead?

Reply:
Changed 👍🏽